docker/tarfile/src.go

```diff
@@ -180,10 +180,12 @@ func (s *Source) loadTarManifest() (*manifestItem, error) {
 	if err := json.Unmarshal(bytes, &items); err != nil {
 		return nil, errors.Wrap(err, "Error decoding tar manifest.json")
 	}
-	if len(items) != 1 {
-		return nil, errors.Errorf("Unexpected tar manifest.json: expected 1 item, got %d", len(items))
-	}
-	return &items[0], nil
+	return items, nil
+}
+
+// LoadTarManifest loads and decodes the manifest.json
+func (s *Source) LoadTarManifest() ([]manifestItem, error) {
```

Comment:
Yup it works fine. I looked it up and a private type can be accessed from an exported function, but the private type can't be created outside its own package.